### DISCUSSION TITLE
Exposing Custom device settings as HA device parameters?

### DISCUSSION THREAD

**Discussion by 3v1n0** [zigpy/zigpy#869] [Q&A] [answered]
Various devices have custom settings (stuff like startup-state in a switch or update timeout in a motion sensor), but in order to configure them it's often needed to go through the zigbee2mqtt device docs to see what's the device support and eventually find the clusters to set in the ZHA device page and configure it (not always trivial either).

So I'm wondering if well known device settings could be exposed as simple switches / entries in the ZHA device page.
It's not clear to me whether this is supported or not.

If that's already supported, should be that handled by exposing device-settings via quirks or what?

**Comment by dmulcahey** [comment, marked as answer]
Already working on this. Several will be exposed in the first major release next year

**Reply by 3v1n0** [reply]
Amazing, will be that handled by zigpy itself or zha quirks will be needed?

**Reply by dmulcahey** [reply]
Most will be pure ZHA
